slurm_ops_manager/slurm_ops.py

```diff
@@ -46,6 +47,7 @@ def __init__(self, charm, component):
         self.infiniband = Infiniband(charm, component)
         if self._slurm_component == "slurmd":
             self.nvidia = NvidiaGPU(charm, component)
+            self.singularity = Singularity(charm, component)
```

Comment:
I'm wondering if we even need to pass the charm and component into the Singularity object. Possibly we change the object to not accept any required initialization parameters.

Reply:
That is needed when the class inherits from `Object`

Reply:
Do we even need to make it a class then?

Reply:
If the class consists of a single install function, possibly we can just define and call a single function?

Reply:
Having a class for that keeps it consistent with having a class for Infiniband and one for Nvidia GPUs. It also makes it easy to add other methods if we need to tweak the settings for Singularity.

We don't _have to_ have a class for it, but it does have some benefits.